internal/controller/rbac/definition/reconciler.go

```diff
@@ -193,11 +196,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			return reconcile.Result{}, err
 		}
 		log.Debug("Applied RBAC ClusterRole")
+		applied = append(applied, cr.GetName())
+	}
+
+	sort.Strings(applied)
+	if len(applied) > 3 {
+		r.record.Event(d, event.Normal(reasonApplyRoles, fmt.Sprintf("Applied RBAC ClusterRoles: %s, %s, %s, and %d more", applied[0], applied[1], applied[2], len(applied)-3)))
+	} else if len(applied) > 0 {
+		r.record.Event(d, event.Normal(reasonApplyRoles, fmt.Sprintf("Applied RBAC ClusterRoles: %s", strings.Join(applied, ", "))))
```

Comment:
Given this pattern is repeated a few times I wonder whether a little utility could help? I'm on the fence. Normally I prefer to avoid such little utilities for things like this, until they're almost everywhere, but I must admit I'm a little (unreasonably?) allergic to `if then else` statements.

I'm thinking something like:

```go
r.record.Event(d, event.Normal(reasonApplyRoles, maybeTruncate("Applied RBAC ClusterRoles", applied)))

// maybeTruncate needs a better name :)
func maybeTruncate(prefix string, names []string) string {
	if len(names) > 3 {
		return fmt.Sprintf("%s: %s, and %d more", prefix, strings.Join(names[:3], ", "), len(names)-3)
	}
	return fmt.Sprintf("%s: %s", prefix, strings.Join(names, ", "))
}
```

Reply:
added. Any good place to put it? Have 3 copies now.

Reply:
I could imagine this being added as a utility in crossplane-runtime somewhere (not sure which package). I'm fine leaving a few copies scattered around for now though - we can figure out a better home if and when it becomes more widely used.